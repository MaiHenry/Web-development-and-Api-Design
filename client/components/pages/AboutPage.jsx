import React from "react";
import "../../Styles.css";

export function AboutPage() {
  return (
    <div className="about-page">
      <h2>About Chatroom</h2>
      <div className="line"></div>
      <p>
        Welcome to Chatroom! Here's a brief of some the possible features in this Application. Just to get you started.
      </p>
      <div className="spacer"></div>
      <div className="feature">
        <h3>Secure platform</h3>
        <p>
            All features are exclusively for users that are logged in. Anonymous users can't navigate further than login.
        </p>
      </div>
      <div className="spacer"></div>
      <div className="feature">
        <h3>Login alternatives</h3>
        <p>
            All users can log in with Google and Microsoft in a seamless fashion.
        </p>
      </div>
      <div className="spacer"></div>
      <div className="feature">
        <h3>Creating Chatrooms</h3>
        <p>
            Microsoft users has an extra benefit of creating chatrooms! Users can also edit and delete their own, and only, chatrooms.
        </p>
      </div>
      <div className="spacer"></div>
      <div className="feature">
        <h3>Your own profile</h3>
        <p>
            All users logged in will have their own user profile with information predefined from the OPEN/ENTRA ID.
            Custom name and biography are available if you want to display yourself in another fashion.
        </p>
      </div>
      <div className="spacer"></div>
      <div className="feature">
        <h3>Chat endlessly in realtime</h3>
        <p>
            All logged in users can join chatrooms and communicate with multiple users from different places, realtime!
            Displayed with either your profilename or custom name, with timestamps.
        </p>
      </div>
      <div className="spacer"></div>
      <div className="feature">
        <h3>All chatrooms are unique</h3>
        <p>
            We don't like redundancy, therefore - no duplicate chatrooms.
        </p>
      </div>
      <div className="spacer"></div>
      <div className="feature">
        <h3>Get to know our chatters!</h3>
        <p>
            Chatroom is a social platform, get to know who's taking part of this event!
            Navigate to the People tab and explore others like urself!
        </p>
      </div>
      <div className="spacer"></div>
      <div className="feature">
        <h3>What happens if you suddenly hit F5?</h3>
        <p>
            Don't worry, you will still be logged in.
        </p>
      </div>
      <div className="spacer"></div>
      <div className="line"></div>
      <div className="feature">
        <h4>At last, thank you for taking your time. <br></br>Wish you a great experience.</h4>
      </div>
    </div>
  );
}
