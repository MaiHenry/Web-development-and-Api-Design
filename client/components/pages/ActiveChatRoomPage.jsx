import React, { useState, useEffect, useContext } from "react";
import { useParams } from "react-router-dom";
import { ApiContext } from "../../ApiContext";
import { LoginContext } from "../../LoginContext";
import { fetchJSON } from "../apiRequests/fetchJSON";
import "../../Styles.css";

export function MessageCard({ message }) {
  const timestamp = new Date(message.timestamp).toLocaleString();
  return (
    <div className="message-card">
      <span className="timestamp">{timestamp}</span>
      <p>
        <strong>{message.name?.profileName}</strong>
      </p>
      <p>{message.content}</p>
    </div>
  );
}

export function ActiveChatRoomPage() {
  const { roomId } = useParams();
  const { user, profileName } = useContext(LoginContext);
  const { fetchMessages, postMessage, fetchChatRoomById } =
    useContext(ApiContext);

  const [roomDetails, setRoomDetails] = useState({});
  const [messages, setMessages] = useState([]);
  const [content, setContent] = useState("");
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState("");
  const [webSocket, setWebSocket] = useState();
  const [customName, setCustomName] = useState("");

  useEffect(() => {
    setIsLoading(true);

    fetchChatRoomById(roomId)
      .then((room) => {
        setRoomDetails(room);
        return fetchMessages(roomId);
      })
      .then((messages) => {
        setMessages(messages);
        setIsLoading(false);
      })
      .catch((err) => {
        console.error("Error:", err);
        setError(err.message);
        setIsLoading(false);
      });

      async function fetchUserData(userEmail) {
        try {
          const userData = await fetchJSON(`/api/login/user/${userEmail}`);
          return userData;
        } catch (error) {
          throw new Error(`Failed to fetch user data: ${error.message}`);
        }
      }
      
      if (user && user.email) {
        fetchUserData(user.email)
          .then((data) => {
            console.log(data);
            setCustomName(data.customName);
            setIsLoading(false);
          })
          .catch((error) => {
            setError(error.message);
            setIsLoading(false);
          });
      } else {
        setIsLoading(false);
        setError("User email is undefined.");
      }

    const webSocket = new WebSocket(
      window.location.origin.replace(/^http/, "ws"),
    );

    webSocket.onmessage = (event) => {
      console.log(event.data);
      setMessages((current) => [...current, JSON.parse(event.data)]);
    };
    setWebSocket(webSocket);
  }, [roomId, fetchChatRoomById, fetchMessages]);

  const handleSendMessage = () => {
    const senderName = customName || profileName || "User";
    const timestamp = new Date().toISOString();
    const messageData = {
      name: { profileName: senderName },
      userId: user?._id,
      userEmail: user.email,
      content,
      timestamp,
    };

    webSocket.send(JSON.stringify(messageData));

    postMessage(roomId, messageData)
      .then(() => {
        setContent("");
        return fetchMessages(roomId);
      })
      .then((updatedMessages) => {
        setMessages(updatedMessages);
      })
      .catch((err) => {
        console.error("Error:", err);
      });
  };

  if (isLoading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return <div>Error: {error}</div>;
  }

  return (
    <div>
      <h2>{roomDetails.name}</h2>
      <p>{roomDetails.description}</p>

      <div className="messages-container">
        {messages.map((message) => (
          <MessageCard key={message._id} message={message} />
        ))}
      </div>

      <div className="message-input">
        <form class="object-form" onSubmit={(e) => e.preventDefault()}>
          <input
            value={content}
            onChange={(e) => setContent(e.target.value)}
            placeholder="Write a message..."
          />
          <button type="submit" onClick={handleSendMessage}>
            Send
          </button>
        </form>
      </div>
    </div>
  );
}