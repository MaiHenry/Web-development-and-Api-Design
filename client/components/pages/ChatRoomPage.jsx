import React, { useState, useEffect, useContext } from "react";
import { useNavigate } from "react-router-dom";
import { ApiContext } from "../../ApiContext";
import { LoginContext } from "../../LoginContext";

function ChatRoomCard({ room, onDelete, onEdit, onEnterRoom }) {
  const { user } = useContext(LoginContext);

  let roomSettings = null;
  if (user && user.email === room.userEmail) {
    roomSettings = (
      <>
        <button onClick={() => onEdit(room._id)}>Edit</button>
        <button onClick={() => onDelete(room._id)}>Delete</button>
      </>
    );
  }

  return (
    <div>
      <h3>{room.name}</h3>
      <p>{room.description}</p>
      <button onClick={() => onEnterRoom(room._id)}>Enter Room</button>
      {roomSettings}
    </div>
  );
}

export function ChatRoomPage() {
  const [chatRooms, setChatRooms] = useState([]);
  const { fetchChatRooms, deleteRoom } = useContext(ApiContext);
  const navigate = useNavigate();
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState(null);
  const { loginMethod } = useContext(LoginContext);

  useEffect(() => {
    setIsLoading(true);
    fetchChatRooms()
      .then((data) => {
        setChatRooms(data);
        setIsLoading(false);
      })
      .catch((err) => {
        setError(err);
        setIsLoading(false);
      });
  }, [fetchChatRooms]);

  if (isLoading) {
    return <div>Loading ...</div>;
  }

  if (error) {
    return <div>Error: {error.message}</div>;
  }

  const handleDeleteRoom = (chatRoomId) => {
    deleteRoom(chatRoomId)
      .then(() => {
        setChatRooms((prevRooms) =>
          prevRooms.filter((room) => room._id !== chatRoomId)
        );
      })
      .catch((error) => {
        console.error("Error:", error);
      });
  };

  const handleCreateRoom = () => {
    navigate("/create");
  };

  function editRoom(chatRoomId) {
    navigate(`/edit/${chatRoomId}`);
  }

  const enterRoom = (chatRoomId) => {
    navigate(`/chatroom/${chatRoomId}`);
  };

  return (
    <div>
      <h2>Available rooms</h2>
      <div className="line"></div>
      <ul>
        {chatRooms.map((room) => (
          <ChatRoomCard
            key={room._id}
            room={room}
            onEnterRoom={enterRoom}
            onEdit={editRoom}
            onDelete={() => handleDeleteRoom(room._id)}
          />
        ))}
      </ul>
      {loginMethod === "microsoft" && (
        <div>
          <div className="spacer"></div>
          <p>Create your own community!</p>
          <div className="spacer"></div>
          <button onClick={handleCreateRoom}>Create new room</button>
        </div>
      )}
    </div>
  );
}
