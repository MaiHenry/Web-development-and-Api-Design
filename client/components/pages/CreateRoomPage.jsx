import React, { useContext, useState } from "react";
import { ApiContext } from "../../ApiContext";
import { useNavigate } from "react-router-dom";
import { LoginContext } from "../../LoginContext";
import "../../Styles.css";

export function CreateRoomPage() {
  const { createRoom } = useContext(ApiContext);
  const navigate = useNavigate();
  const { user } = useContext(LoginContext)

  const [name, setName] = useState("");
  const [description, setDescription] = useState("");
  const [successMessage, setSuccessMessage] = useState("");
  const [errorMessage, setErrorMessage] = useState("");

  function handleSubmit(event) {
    event.preventDefault();

    const roomData = {
      name,
      description,
      userId: user && user._id // Room connected to logged in user.
    };

    createRoom(roomData)
      .then(() => {
        setSuccessMessage("Room created!");
        navigate("/chatroom");
      })
      .catch((error) => {
        setErrorMessage("Could not add the room.");
        console.error("Error adding room", error);
      });
  }

  return (
    <form className="object-form" onSubmit={handleSubmit}>
      <h2>Create a Chat Room</h2>

      <div>
        <label>Name:</label>
        <input
          required
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
      </div>

      <div>
        <label>Description:</label>
        <textarea
          required
          value={description}
          onChange={(e) => setDescription(e.target.value)}
        />
      </div>

      <button type="submit">Create Room</button>
      {successMessage && <div>{successMessage}</div>}
      {errorMessage && <div>{errorMessage}</div>}
    </form>
  );
}
