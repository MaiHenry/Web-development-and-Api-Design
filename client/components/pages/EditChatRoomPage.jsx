import React, { useState, useEffect, useContext } from "react";
import { useNavigate, useParams } from "react-router-dom";
import { ApiContext } from "../../ApiContext";
import { LoginContext } from "../../LoginContext";

export function EditChatRoomPage() {
  const { fetchChatRoomById, updateChatRoom } = useContext(ApiContext);
  const { user } = useContext(LoginContext);
  const navigate = useNavigate();
  const { chatRoomId } = useParams();

  const [name, setName] = useState("");
  const [description, setDescription] = useState("");

  useEffect(() => {
    if (chatRoomId) {
      fetchChatRoomById(chatRoomId)
        .then((room) => {
          setName(room.name);
          setDescription(room.description);
        })
        .catch((error) => {
          console.error("Error getting chat room:", error);
        });
    }
  }, [chatRoomId, fetchChatRoomById]);

  const handleSubmit = async (event) => {
    event.preventDefault();
    try {
      await updateChatRoom(chatRoomId, {
        name,
        description,
        userEmail: user.email,
      });
      navigate("/");
    } catch (error) {
      console.error("Error updating room:", error);
    }
  };

  return (
    <form className="object-form" onSubmit={handleSubmit}>
      <h2>Edit Room</h2>
      <div>
        <label>Name:</label>
        <input
          type="text"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
      </div>
      <div>
        <label>Description:</label>
        <textarea
          value={description}
          onChange={(e) => setDescription(e.target.value)}
        />
      </div>
      <button type="submit">Update Room</button>
    </form>
  );
}