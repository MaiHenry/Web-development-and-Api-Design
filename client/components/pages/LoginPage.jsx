import React from "react";
import { LoginWithOpenIdButton } from "../login/LoginWithOpenIdButton";
import '../../Styles.css';  

export function LoginPage() {

  return (
    <>
      <h2>Choose</h2>
      <div className="headerDivider"></div>
      <LoginWithOpenIdButton />
      <div className="spacer"></div>
    </>
  );
}
