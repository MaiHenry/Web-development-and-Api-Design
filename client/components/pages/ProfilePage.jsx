import React, { useContext, useState } from "react";
import { useNavigate } from "react-router-dom";
import { LoginContext } from "../../LoginContext";
import { putJSON } from "../apiRequests/putJSON";
import "../../Styles.css";

export function ProfilePage() {
  const navigate = useNavigate();
  const { user, profileName, unloadUser, customName, customBio, loadUser } =
    useContext(LoginContext);
  const [newCustomName, setNewCustomName] = useState(customName);
  const [newCustomBio, setNewCustomBio] = useState(customBio);
  const [updateSuccess, setUpdateSuccess] = useState(false);

  const handleUpdate = async (e) => {
    e.preventDefault();
    try {
      await putJSON(`/api/login/user/update/${user.email}`, {
        customName: newCustomName,
        customBio: newCustomBio,
      });
      loadUser();
      await putJSON(`/api/messages/${user.email}`, {
        newCustomName: newCustomName,
      });

      setUpdateSuccess(true);
      setNewCustomName(newCustomName);
      setNewCustomBio(newCustomBio);
    } catch (error) {
      console.error("Failed to update user:", error);
    }
  };

  async function handleSubmitLogout(e) {
    e.preventDefault();
    const res = await fetch("/api/login", {
      method: "DELETE",
    });
    if (!res.ok) {
      throw new Error("Failed to log out " + res.statusText);
    }
    unloadUser();
    navigate("/");
  }

  if (!user) {
    return <p>Loading...</p>; // handle for not logged in state
  }

  return (
    <div className="profile-container">
      <h2>Profile Page</h2>
      <div className="line"></div>
      <div className="profile-info">
        {user.picture === "https://graph.microsoft.com/v1.0/me/photo/$value" ||
        user.picture == null ? (
          <p>
            Welcome! <br></br>
          </p> // Placeholder in case of no picture
        ) : (
          <div className="profile-picture">
            <img src={user.picture} alt="Profile" />
          </div>
        )}
        <p>
          <strong>Name:</strong> {profileName}
        </p>
        <p>
          <strong>Address:</strong> {user.email}
        </p>
        <p>
          <strong>Custom Name:</strong> {customName || ""}
        </p>
        <p>
          <strong>Custom Bio:</strong> {customBio || ""}
        </p>
        <div className="spacer"></div>
        <div className="line"></div>
        <h3>Update your custom name and bio!</h3>
        <p>Changes will affect your past, present and future name</p>
        <div className="spacer"></div>
        <form onSubmit={handleUpdate} className="profile-form">
          <div>
            <input
              type="text"
              value={newCustomName}
              onChange={(e) => setNewCustomName(e.target.value)}
              placeholder="Custom Name"
              required
            />
          </div>
          <div>
            <textarea
              rows="5"
              value={newCustomBio}
              onChange={(e) => setNewCustomBio(e.target.value)}
              placeholder="Custom Bio"
            />
          </div>
          <button type="submit">Update Profile</button>
          {updateSuccess && <p>Your changes was successful!</p>}
          <div className="spacer"></div>
        </form>
      </div>

      <form onSubmit={handleSubmitLogout} className="logout-form">
        <button>Log out</button>
      </form>
    </div>
  );
}
