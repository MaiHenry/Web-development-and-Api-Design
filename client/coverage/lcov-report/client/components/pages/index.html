
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for client/components/pages</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> client/components/pages</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">45.19% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>47/104</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>16/32</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">24.32% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/37</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">45.63% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>47/103</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="AboutPage.jsx"><a href="AboutPage.jsx.html">AboutPage.jsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file low" data-value="ChatRoomPage.jsx"><a href="ChatRoomPage.jsx.html">ChatRoomPage.jsx</a></td>
	<td data-value="48.71" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 48%"></div><div class="cover-empty" style="width: 52%"></div></div>
	</td>
	<td data-value="48.71" class="pct low">48.71%</td>
	<td data-value="39" class="abs low">19/39</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="18" class="abs low">4/18</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="38" class="abs medium">19/38</td>
	</tr>

<tr>
	<td class="file low" data-value="CreateRoomPage.jsx"><a href="CreateRoomPage.jsx.html">CreateRoomPage.jsx</a></td>
	<td data-value="47.05" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 47%"></div><div class="cover-empty" style="width: 53%"></div></div>
	</td>
	<td data-value="47.05" class="pct low">47.05%</td>
	<td data-value="17" class="abs low">8/17</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="16.66" class="pct low">16.66%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="47.05" class="pct low">47.05%</td>
	<td data-value="17" class="abs low">8/17</td>
	</tr>

<tr>
	<td class="file low" data-value="EditChatRoomPage.jsx"><a href="EditChatRoomPage.jsx.html">EditChatRoomPage.jsx</a></td>
	<td data-value="47.61" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 47%"></div><div class="cover-empty" style="width: 53%"></div></div>
	</td>
	<td data-value="47.61" class="pct low">47.61%</td>
	<td data-value="21" class="abs low">10/21</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="47.61" class="pct low">47.61%</td>
	<td data-value="21" class="abs low">10/21</td>
	</tr>

<tr>
	<td class="file low" data-value="ProfilePage.jsx"><a href="ProfilePage.jsx.html">ProfilePage.jsx</a></td>
	<td data-value="34.61" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 34%"></div><div class="cover-empty" style="width: 66%"></div></div>
	</td>
	<td data-value="34.61" class="pct low">34.61%</td>
	<td data-value="26" class="abs low">9/26</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="14" class="abs medium">8/14</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="34.61" class="pct low">34.61%</td>
	<td data-value="26" class="abs low">9/26</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-11-23T05:36:29.627Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    