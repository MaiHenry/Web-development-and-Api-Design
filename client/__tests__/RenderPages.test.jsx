import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { CreateRoomPage } from "../components/pages/CreateRoomPage";
import { ChatRoomPage } from "../components/pages/ChatRoomPage";
import { AboutPage } from "../components/pages/AboutPage";
import { ActiveChatRoomPage } from "../components/pages/ActiveChatRoomPage";
import { EditChatRoomPage } from "../components/pages/EditChatRoomPage";
import { ProfilePage } from "../components/pages/ProfilePage";

test("render AboutPage", () => {
    const component = renderer.create(
      <MemoryRouter>
        <AboutPage />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test("render ChatRoomPage", () => {
    const component = renderer.create(
      <MemoryRouter>
        <ChatRoomPage />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test("render CreateRoomPage", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CreateRoomPage />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test("render EditChatRoomPage", () => {
    const component = renderer.create(
      <MemoryRouter>
        <EditChatRoomPage />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test("render ProfilePage", () => {
      const component = renderer.create(
        <MemoryRouter>
          <ProfilePage />
        </MemoryRouter>
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  
  