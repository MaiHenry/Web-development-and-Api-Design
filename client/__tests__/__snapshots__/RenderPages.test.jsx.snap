// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AboutPage 1`] = `
<div
  className="about-page"
>
  <h2>
    About Chatroom
  </h2>
  <div
    className="line"
  />
  <p>
    Welcome to Chatroom! Here's a brief of some the possible features in this Application. Just to get you started.
  </p>
  <div
    className="spacer"
  />
  <div
    className="feature"
  >
    <h3>
      Secure platform
    </h3>
    <p>
      All features are exclusively for users that are logged in. Anonymous users can't navigate further than login.
    </p>
  </div>
  <div
    className="spacer"
  />
  <div
    className="feature"
  >
    <h3>
      Login alternatives
    </h3>
    <p>
      All users can log in with Google and Microsoft in a seamless fashion.
    </p>
  </div>
  <div
    className="spacer"
  />
  <div
    className="feature"
  >
    <h3>
      Creating Chatrooms
    </h3>
    <p>
      Microsoft users have an extra benefit of creating chatrooms! Users can also edit and delete their own, and only, chatrooms.
    </p>
  </div>
  <div
    className="spacer"
  />
  <div
    className="feature"
  >
    <h3>
      Your own profile
    </h3>
    <p>
      All users logged in will have their own user profile with information predefined from the OPEN/ENTRA ID. Custom name and biography are available if you want to display yourself in another fashion.
    </p>
  </div>
  <div
    className="spacer"
  />
  <div
    className="feature"
  >
    <h3>
      Chat endlessly in realtime
    </h3>
    <p>
      All logged in users can join chatrooms and communicate with multiple users from different places, realtime! Displayed with either your profilename or custom name, with timestamps.
    </p>
  </div>
  <div
    className="spacer"
  />
  <div
    className="feature"
  >
    <h3>
      All chatrooms are unique
    </h3>
    <p>
      We don't like redundancy, therefore - no duplicate chatrooms.
    </p>
  </div>
  <div
    className="spacer"
  />
  <div
    className="feature"
  >
    <h3>
      Get to know our chatters!
    </h3>
    <p>
      Chatroom is a social platform, get to know who's taking part of it! Navigate to the People tab and explore other contributors!
    </p>
  </div>
  <div
    className="spacer"
  />
  <div
    className="feature"
  >
    <h3>
      What happens if you suddenly hit F5 or accidentally refresh?
    </h3>
    <p>
      Don't worry, you will still be logged in.
    </p>
  </div>
  <div
    className="spacer"
  />
  <div
    className="line"
  />
  <div
    className="feature"
  >
    <h4>
      At last, thank you for taking your time. 
      <br />
      Wish you a great experience.
    </h4>
  </div>
</div>
`;

exports[`render ChatRoomPage 1`] = `
<div>
  Loading ...
</div>
`;

exports[`render CreateRoomPage 1`] = `
<form
  className="object-form"
  onSubmit={[Function]}
>
  <h2>
    Create a Chat Room
  </h2>
  <div>
    <label>
      Name:
    </label>
    <input
      onChange={[Function]}
      required={true}
      value=""
    />
  </div>
  <div>
    <label>
      Description:
    </label>
    <textarea
      onChange={[Function]}
      required={true}
      value=""
    />
  </div>
  <div
    className="line"
  />
  <button
    type="submit"
  >
    Create Room
  </button>
</form>
`;

exports[`render EditChatRoomPage 1`] = `
<form
  className="object-form"
  onSubmit={[Function]}
>
  <h2>
    Edit Room
  </h2>
  <div>
    <label>
      Name:
    </label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label>
      Description:
    </label>
    <textarea
      onChange={[Function]}
      value=""
    />
  </div>
  <button
    type="submit"
  >
    Update Room
  </button>
</form>
`;

exports[`render ProfilePage 1`] = `
<p>
  Loading...
</p>
`;
